### ISSUE TITLE
Get other sublocation

### ISSUE THREAD

**Issue by alasta** [zscaler/terraform-provider-zia#265]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v1.2.2

### Affected Resource(s)

- Data source zia_sub_location_management

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
data "zia_location_management" "gre_other"{
    name = "other"
    parent_name = "MY_LOC"
    }
```
note : I  am trying with "zia_sub_location_management" but don't exist.

### Debug Output

When I use this in location of FW rule :
```
│ Error: Null value found in list
│ 
│   with module.fw_rule.zia_firewall_filtering_rule.this,
│   on .terraform/modules/fw_rule/main.tf line 1, in resource "zia_firewall_filtering_rule" "this":
│    1: resource "zia_firewall_filtering_rule" "this" {
│ 
│ Null values are not allowed for this attribute value.
```

### Steps to Reproduce

1. `terraform apply`

**Comment by willguibr** [comment]
@alasta 
The resource is working as designed/expected. Please make sure you are using the most recent version of the provider [v2.6.1](https://github.com/zscaler/terraform-provider-zia/releases/tag/v2.6.1) that supports this capability.
To retrieve the ID of a sub-location use the following datasource:

```hcl
data "zia_location_management" "this"{
    parent_name = "MY_LOC"
    name = "other"
}
```
To associate the retrieved sub-location ID for "other" sub-location with a Firewall rule or any other resource that accepts a location as a criteria, use the following resource structure as per the documentation [here](https://registry.terraform.io/providers/zscaler/zia/latest/docs/resources/zia_firewall_filtering_rule).

```hcl
resource "zia_firewall_filtering_rule" "example265" {
    name = "Example265"
    description = "Example265"
    action = "ALLOW"
    state = "ENABLED"
    order = 10
    enable_full_logging = true
    locations {
        id = [ data.zia_location_management.this.id ]
    }
}
```
Let us know if you have any problems.
Thanks

**Comment by alasta** [comment]
Hi William,

I will test that tomorrow and let you if I have a problems.
Thank you.

**Comment by alasta** [comment]
Hi  @willguibr,

Same problem with the provider v2.6.1 when I add the data source to other in locations id.

Configuration 
```hcl
data "zia_location_management" "gre_other"{
    name = "other"
    parent_name = "MY_LOC" 
    depends_on = [ module.location_ratp_gre_on_premise_loop ]
}

module "fw_rule_deny_servers" {
  source       = "git::https://...../terraform-zscaler-zia-firewall-filtering-rule.git?ref=v0.1.0"
  name         = "SERVER"
  description  = "Block reset servers"
  action       = "BLOCK_RESET"
  state        = "ENABLED"
  order        = 7
  locations_id = [module.sub_location_gre_on_premise_servers_001.id, module.sub_location_gre_on_premise_servers_002.id, data.zia_location_management.gre_other.id ]
  depends_on = [
    data.zia_location_management.gre_other
  ]
}
```

The Terraform Plan send this error :
```hcl
module.fw_rule_deny_servers.zia_firewall_filtering_rule.this: Refreshing state... [id=733092]
╷
│ Error: Null value found in list
│ 
│   with module.fw_rule_deny_servers.zia_firewall_filtering_rule.this,
│   on .terraform/modules/fw_rule_deny_servers/main.tf line 1, in resource "zia_firewall_filtering_rule" "this":
│    1: resource "zia_firewall_filtering_rule" "this" {
│ 
│ Null values are not allowed for this attribute value.
╵
```
Best regards

**Comment by willguibr** [comment]
This has been addressed in v2.6.2
